tests/test_make.py

```diff
@@ -1250,6 +1251,25 @@ class C(object):
 
         assert C() == copy.deepcopy(C())
 
+    def test_no_references_to_original(self):
+        """
+        When subclassing a slots class, there are no stray references to the
+        original class.
+        """
+
+        @attr.s(slots=True)
+        class C(object):
+            pass
+
+        @attr.s(slots=True)
+        class C2(C):
+            pass
+
+        # The original C2 is in a reference cycle, so force a collect:
+        gc.collect()
```

Comment:
Sometimes calling `gc.collect()` once isn't enough to actually collect all garbage:
https://github.com/python-trio/trio/blob/45b4e586897f2676169b89e6b9f48d54eb14621a/trio/_core/tests/tutil.py#L26-L37
In this case I think it is probably fine but generally I feel like by the time you're calling `gc.collect` in a test you're already having a bad day and don't want to get surprised by any *new* edge cases.

Reply:
That's interesting; I wasn't aware of that. IIUC, we'll be OK here because there are no `__del__` methods involved; does that sound right to you?

Reply:
I was going to say "yes", but actually... How certain are we that there's no `__del__` involved in clearing the parent's subclasses table? (I suppose it probably uses weakrefs rather than `__del__`, but I'm not certain, and I'm not certain how weakrefs interact with `gc.collect` anyway.)